server/src/test/java/org/elasticsearch/index/shard/IndexShardOperationPermitsTests.java

```diff
@@ -716,4 +703,22 @@ public void testPermitTraceCapturing() throws ExecutionException, InterruptedExc
         assertThat(permits.getActiveOperationsCount(), equalTo(0));
         assertThat(permits.getActiveOperations(), emptyIterable());
     }
+
+    private static ActionListener<Releasable> wrap(final CheckedRunnable<Exception> onResponse) {
+        return new ActionListener<Releasable>() {
+            @Override
+            public void onResponse(final Releasable releasable) {
+                try (Releasable ignored = releasable) {
+                    onResponse.run();
+                } catch (final Exception e) {
+                    onFailure(e);
+                }
+            }
+
+            @Override
+            public void onFailure(final Exception e) {
+                throw new RuntimeException(e);
```

Comment:
not sure I follow what you are doing here. Why do you re-throw? can this situation happen actually?

Reply:
maybe you want an new AssertionError(e) instead?

Reply:
Thanks for catching this - new AssertionError(e) indeed